packages/react-i18n/src/babel-plugin/babel-templates.ts

```diff
@@ -89,3 +96,62 @@ export function i18nGeneratedDictionaryCallExpression(
     },
   )();
 }
+
+function buildExplicitImports(bindingId: string, localeIds: string[]) {
+  if (localeIds.length === 0) {
+    return `return {};`;
+  } else if (localeIds.length === 1) {
+    return `return ${generateImportDefinition(
+      {id: localeIds[0], explicit: true},
+      bindingId,
+    )}${generateThenChain(generateReturnDefaultFunction())};`;
+  }
+
+  const returnDefaultVariable = 'returnDefault';
+  const generateReturnStatement = (id: string) => {
+    return `return ${generateImportDefinition(
+      {id, explicit: true},
+      bindingId,
+    )}${generateThenChain(returnDefaultVariable)};`;
+  };
+
+  const switchStatementInternals = localeIds.reduce((merged, value) => {
+    return merged.concat(`case "${value}":
+      ${generateReturnStatement(value)}
+    `);
+  }, '');
+
+  const localeSelectionStatement = `switch (locale) {
+    ${switchStatementInternals}
+  }`;
+
+  return `const ${returnDefaultVariable} = ${generateReturnDefaultFunction()};
+  ${localeSelectionStatement}`;
+}
+
+function generateImportDefinition(
+  file: {
+    id: string;
+    explicit: boolean;
+  },
+  bindingId: string,
+) {
+  const isExplicit = file.explicit;
+  const fileName = isExplicit ? file.id : `\${${file.id}}`;
+
+  const fileLocation = `./${TRANSLATION_DIRECTORY_NAME}/${fileName}.json`;
+  const wrappedLocation = isExplicit
+    ? `"${fileLocation}"`
+    : `\`${fileLocation}\``;
+  const webpackMode = isExplicit ? '' : ', webpackMode: "lazy-once"';
+
+  return `import(/* webpackChunkName: "${bindingId}-i18n"${webpackMode} */ ${wrappedLocation})`;
+}
```

Comment:
I find this block of logic difficult to understand, how about simply:

```suggestion
function generateImportDefinition(
  bindingId: string,
  explicitFileName?: string,
) {
  if (explicitFileName) {
    return `import(/* webpackChunkName: "${bindingId}-i18n" */ "./${TRANSLATION_DIRECTORY_NAME}/${explicitFileName}.json")`;
  }

  return `import(/* webpackChunkName: "${bindingId}-i18n", webpackMode: "lazy-once" */ \`./${TRANSLATION_DIRECTORY_NAME}/\${locale}.json\`)`;
}
```